nemo_curator/download/commoncrawl.py

```diff
@@ -71,55 +73,126 @@ def lang_detect(decoded_html):
+            # Return nothing when we cannot segment the document
+            if logger is not None:
+                logger.info("Could not segment paragaphs in the document")
+            return
+        paragraphs = handler.paragraphs
+
+        # Context free classification
+        justext.core.classify_paragraphs(
+            paragraphs,
+            stop_words,
+            length_low,
+            length_high,
+            stopwords_low,
+            stopwords_high,
+            max_link_density,
+            no_headings,
+        )
 
-    # Copy the context free class to the class_style
-    # This handles the headings as described in the
-    # documentation
-    for paragraph in paragraphs:
-        paragraph.class_type = paragraph.cf_class
+        # Copy the context free class to the class_style
+        # This handles the headings as described in the
+        # documentation
+        for paragraph in paragraphs:
+            paragraph.class_type = paragraph.cf_class
 
-    # Context sensitive classification
-    justext.core.revise_paragraph_classification(
-        paragraphs,
-        max_heading_distance,
-    )
+        # Context sensitive classification
+        justext.core.revise_paragraph_classification(
+            paragraphs,
+            max_heading_distance,
+        )
 
-    return [p.text for p in paragraphs if not p.is_boilerplate]
+        return [p.text for p in paragraphs if not p.is_boilerplate]
+    elif algorithm.lower() == "resiliparse":
+        if length_low is not None:
+            warnings.warn(
+                "jusText-specific parameter 'length_low' is set but will be ignored when using Resiliparse.",
+                UserWarning,
+            )
+        if length_high is not None:
+            warnings.warn(
+                "jusText-specific parameter 'length_high' is set but will be ignored when using Resiliparse.",
+                UserWarning,
+            )
+        if stopwords_low is not None:
+            warnings.warn(
+                "jusText-specific parameter 'stopwords_low' is set but will be ignored when using Resiliparse.",
+                UserWarning,
+            )
+        if max_link_density is not None:
+            warnings.warn(
+                "jusText-specific parameter 'max_link_density' is set but will be ignored when using Resiliparse.",
+                UserWarning,
+            )
+        if max_heading_distance is not None:
+            warnings.warn(
+                "jusText-specific parameter 'max_heading_distance' is set but will be ignored when using Resiliparse.",
+                UserWarning,
+            )
+        if no_headings is not None:
+            warnings.warn(
+                "jusText-specific parameter 'no_headings' is set but will be ignored when using Resiliparse.",
+                UserWarning,
+            )
+
+        text = extract_plain_text(html, main_content=True, alt_texts=False)
```

Comment:
Probably best to pass these arguments and the remaining ones in the function on to the user.